src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3922,6 +3917,9 @@ int journal_file_copy_entry(JournalFile *from, JournalFile *to, Object *o, uint6
                 } else
                         data = o->data.payload;
 
+                if (l == 0)
+                        continue;
```

Comment:
should fail with -EBADMSG, no?

Reply:
I figured there could potentially be old journals out there with zero sized data that we wouldn't want to fail on. Kind of a moot point though since journal_file_copy_entry() is only called to flush from runtime to persistent journal. Will change.